pdf/internal/e2etest/passthrough_test.go

```diff
@@ -0,0 +1,177 @@
+/*
+ * This file is subject to the terms and conditions defined in
+ * file 'LICENSE.md', which is part of this source code package.
+ */
+
+package e2etest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/unidoc/unidoc/common"
+	"github.com/unidoc/unidoc/pdf/model"
+)
+
+// Passthrough benchmark loads a PDF, writes back out and performs a sanity check on the output with ghostscript.
+// Set environment variables:
+//		UNIDOC_E2E_FORCE_TESTS to "1" to force the tests to execute.
+//		UNIDOC_PASSTHROUGH_TESTDATA to the path of the corpus folder.
+//		UNIDOC_GS_BIN_PATH to the path of the ghosccript binary (gs).
+var (
+	forceTest               = os.Getenv("UNIDOC_E2E_FORCE_TESTS") == "1"
+	passthroughCorpusFolder = os.Getenv("UNIDOC_PASSTHROUGH_TESTDATA")
+)
+
+func TestPassthrough(t *testing.T) {
+	if len(passthroughCorpusFolder) == 0 {
+		if forceTest {
+			t.Fatalf("UNIDOC_PASSTHROUGH_TESTDATA not set")
+		}
+	}
+
+	files, err := ioutil.ReadDir(passthroughCorpusFolder)
+	if err != nil {
+		if forceTest {
+			t.Fatalf("Error opening %s: %v", passthroughCorpusFolder, err)
+		}
+		t.Skipf("Skipping passthrough bench - unable to open UNIDOC_PASSTHROUGH_TESTDATA (%s)", passthroughCorpusFolder)
+		return
+	}
+
+	// Make a temporary folder and clean up after.
+	tempdir, err := ioutil.TempDir("", "unidoc_passthrough")
+	if err != nil {
+		t.Fatalf("Failed to create temporary folder")
+	}
+	defer os.RemoveAll(tempdir)
+
+	for _, file := range files {
+		t.Logf("%s", file.Name())
+		fpath := filepath.Join(passthroughCorpusFolder, file.Name())
+		params := passthroughParams{
+			inputPath:    fpath,
+			outPath:      filepath.Join(tempdir, "1.pdf"),
+			gsValidation: false,
```

Comment:
Should not there be a way to enable Ghostscript validation? gsValidation is always false here.